### PR TITLE
para-registrar-swap

### PR DESCRIPTION
**Pull request by girazoki**
para-registrar-swap script

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,7 +123,7 @@ The script accepts these inputs fields:
 ### Example to note Pre-Image and propose
 `yarn set-transact-info --ws-provider ws://127.0.0.1:34102  --destination  '{ "parents": 1, "interior": "Here" }' --fee-per-second 8 --extra-weight 3000000000 --max-weight 20000000000 --account-priv-key "0x5fb92d6e98884f76de468fa3f6278f8807c48bebc13595d45af5bdc4da702133" --send-preimage-hash true --send-proposal-as  democracy`
 
-### Example to note Pre-Image and propose through council
+### Example to note Pre-Image and propse through council
```
Comment (jboetticher) [inline README.md:126]:
```suggestion
### Example to note Pre-Image and propose through council
```

File: README.md [hunk 2]
```diff
@@ -205,3 +205,18 @@ The script accepts these inputs fields:
 ### Example through democracy but batching 2 txs
 
 `yarn generic-call-propose -w ws://127.0.0.1:34102  --call "0x0302f24ff3a9cf04c71dbc94d0b566f7a27b94566cacc0f0f4ab324c46e55d02d0033343b4be8a55532d28" --call "0x0302f24ff3a9cf04c71dbc94d0b566f7a27b94566cacc0f0f4ab324c46e55d02d0033343b4be8a55532d28" --account-priv-key "0x5fb92d6e98884f76de468fa3f6278f8807c48bebc13595d45af5bdc4da702133" --send-preimage-hash true --send-proposal-as democracy`
+
+## Para-registrar-swap
+
+Script that allows to do a para id swap in the relay from the parachain.
```
Comment (jboetticher) [inline README.md:211]:
```suggestion
Script that allows a para id swap in the relay from the parachain.
```

File: scripts/para-registrar-swapper.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+ 
+// Construct
+const wsProvider = new WsProvider(args['parachain-ws-provider']);
+const relayProvider = new WsProvider(args['relay-ws-provider']);
+
+async function main () {
+    const api = await ApiPromise.create({ provider: wsProvider });
+    const relayApi = await ApiPromise.create({ provider: relayProvider });
+
+    const proposalAmount = await api.consts.democracy.minimumDeposit as any;
```
Comment (jboetticher) [inline scripts/para-registrar-swapper.ts:31]:
```suggestion
```

File: scripts/para-registrar-swapper.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    console.log("Encoded proposal hash for batch utility after schedule is %s", encodedHash);
+    console.log("Encoded length %d", encodedProposal.length);
+
+    if (args["send-preimage-hash"]) {
+        await api.tx.democracy
+        .notePreimage(encodedProposal)
+        .signAndSend(account, { nonce: nonce++ });
+
+        if (args["send-proposal-as"] == 'democracy') {
+            await api.tx.democracy
```
Comment (jboetticher) [inline scripts/para-registrar-swapper.ts:105]:
```suggestion
            const proposalAmount = await api.consts.democracy.minimumDeposit as any;
            await api.tx.democracy
```

**Review by jboetticher — APPROVED** [review]

**Review by albertov19 — COMMENTED** [review]
@jboetticher I would've adapted this script so that it follows the same as others of using the helper functions.

The idea is that becomes a standard. Also, for the future, if you push some commits you should request review from others :)

**Pull request merged**